File: app/in/partake/controller/api/debug/InvalidAPI.java
package in.partake.controller.api.debug;

import play.mvc.Result;
import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.dao.DAOException;
import in.partake.resource.UserErrorCode;

public class InvalidAPI extends AbstractPartakeAPI {

    public static Result get() throws DAOException, PartakeException {
        return new InvalidAPI().execute();
    }

    /**
     * 常に <code>{ "result": "error", "reason": "intentional invalid response" }</code> を返す。
     * ステータスは 400 を返す。
     */
    @Override
    protected Result doExecute() throws DAOException {
        return renderInvalid(UserErrorCode.INTENTIONAL_USER_ERROR);
    }
}

File: app/controllers/api/debug/InvalidAPI.scala
package controllers.api.debug

import controllers.api.AbstractAPI
import controllers.ActionContext
import in.partake.resource.UserErrorCode
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult

object InvalidAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    return renderInvalid(UserErrorCode.INTENTIONAL_USER_ERROR);
  }
}
